### PR TITLE
Document structs, maps, keyword lists via testing & fix accidental key replacement bug

### PR DESCRIPTION
**Pull request by angelikatyborska**
This PR:
- Turns all expected representations in tests into Elixir files so that we get syntax coloring.
- Adds a bunch of new tests for structs, maps, and keyword lists to document the current behavior.
- Fixes a bug where keys in structs, maps, or keyword lists would be replaced with a placeholder if they happened to have the same name as a previously used variable somewhere in the code.

### CODE REVIEW

File: lib/representer.ex [hunk 1]
```diff
@@ -277,6 +277,47 @@ defmodule Representer do
...
+
+        value ->
+          {value, represented}
+      end
+
+    {{:@, meta, [{name, meta2, value}]}, represented}
+  end
+
+  # TODO: `when` clauses in specs but only for variable names, not key names in maps
+
```
Comment (angelikatyborska) [inline lib/representer.ex:320]: This seems a bit crazy already and it's not the full fix yet 🙊 I'm trying to fix the consequences of [`d2fc2d3` (#73)](https://github.com/exercism/elixir-representer/pull/73/commits/d2fc2d3df8e289b266c1c831729f4200df0a999b) but I'm not sure if it's worth it.

Comment (jiegillet) [reply lib/representer.ex:320]: Haha yeah it does seem a bit crazy and endless.
Representers get so complex after a while :(

File: test_data/user_defined_structs/expected_representation.ex [hunk 2]
```diff
@@ -0,0 +1,24 @@
+defmodule Placeholder_1 do
+  defstruct [:wheels, :doors, :color]
```
Comment (jiegillet) [inline test_data/user_defined_structs/expected_representation.ex:2]: Eventually, we will want these user defined keys to be replaced as mentionned in #37, right?

**Review by jiegillet — APPROVED** [review]
OK, it works as expected, but the special cases are a bit frightening.

File: test_data/user_defined_structs/expected_representation.ex (hunk 2 above)
Comment (angelikatyborska) [reply test_data/user_defined_structs/expected_representation.ex:2]: Yes

**Pull request merged**